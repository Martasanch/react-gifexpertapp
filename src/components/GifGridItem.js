import React from 'react'
import PropTypes from "prop-types";

export const GifGridItem = ({id, title, url}) => {
//console.log(title, url)
    return (
        <div className='card animate__animated animate__bounce animate__fadeIn'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}


//Para obligar a poner un valor a las prop
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };
 