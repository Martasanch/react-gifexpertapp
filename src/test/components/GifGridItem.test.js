import react from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"



describe('Pruebas en GifGridItem', () => {
    const title = 'TÃ­tulo de prueba'
    const url = 'http://localhost/prueba.jpg'

    test('Debe de mostar el componente correctamente', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url} />)
        expect( wrapper).toMatchSnapshot()
    })
})



