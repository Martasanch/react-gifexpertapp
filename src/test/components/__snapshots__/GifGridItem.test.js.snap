// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Título de prueba"
    src="http://localhost/prueba.jpg"
  />
  <p>
    Título de prueba
  </p>
</div>
`;
